### PR TITLE
[automation] Improve build

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Doc build and version updates should now automatically add new ones when run, which will be a lot simpler.

### Test Plan
Ran locally

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -118,3 +118,8 @@ and every commit is able to be built and passes all checks performed by CI.
 
 The Aptos SDK uses [GitHub issues](https://github.com/aptos-labs/aptos-ts-sdk/issues) to track bugs. Please include 
 necessary information and instructions to reproduce your issue.
+
+## Releasing a new version
+
+Simply update the version in `package.json` and `pnpm update-version` will take care of the rest. This will create a new
```
Comment (0xmaayan) [inline CONTRIBUTING.md:124]: lets also mention checking out to a new branch before updating/running the command

File: scripts/generateDocs.sh [hunk 2]
```diff
@@ -1,4 +1,20 @@
+#!/bin/sh
+
 # Generate docs by version for the SDK
 # References the Github repo directly
 # shellcheck disable=SC2154
 cross-var npx typedoc src/index.ts --out "docs/@aptos-labs/ts-sdk-$npm_package_version" --cleanOutputDir --excludeInternal --includeVersion
+
+# Update the main page
+INDEX_FILE='docs/index.md'
+FIRST_LINE=7
```
Comment (0xmaayan) [inline scripts/generateDocs.sh:10]: why 7?

File: scripts/generateDocs.sh [hunk 3]
```diff
@@ -1,4 +1,20 @@
...
+NEXT_LINE=$((FIRST_LINE + 1))
+
+# Add in the middle line the new version.
+# TODO: Make this more stable / a different way of generating that isn't just inserting a line
+{
+  head -n $FIRST_LINE $INDEX_FILE;
+  cross-var echo "- [Stable - @aptos-labs/ts-sdk-$npm_package_version](@aptos-labs/ts-sdk-$npm_package_version)";
```
Comment (0xmaayan) [inline scripts/generateDocs.sh:17]: for any non-stable version, this is a manual process - right?

**Review by 0xmaayan — APPROVED** [review]
noice!

File: scripts/generateDocs.sh (hunk 3 above)
Comment (gregnazario) [reply scripts/generateDocs.sh:17]: Yes, I can also just add a label option

**Pull request merged**